File: javascript/allergies/allergies.js
const Allergies = function(flags) {
  this.flags = flags;
};

Allergies.allergens = [
  "eggs",
  "peanuts",
  "shellfish",
  "strawberries",
  "tomatoes",
  "chocolate",
  "pollen",
  "cats"
];

Allergies.prototype.list = function() {
  return Allergies.allergens.filter((k) => this.allergicTo(k));
};

Allergies.prototype.allergicTo = function(item) {
  return (this.flags & 1 << Allergies.allergens.indexOf(item)) != 0;
};

module.exports = Allergies;

File: python/allergies/allergies.py
class Allergies(object):

    _allergens = [
        'eggs',
        'peanuts',
        'shellfish',
        'strawberries',
        'tomatoes',
        'chocolate',
        'pollen',
        'cats'
    ]

    def __init__(self, flags):
        self.flags = flags

    def is_allergic_to(self, item):
        return self.flags & 1 << self._allergens.index(item)

    @property
    def lst(self):
        return filter(lambda a: self.is_allergic_to(a), self._allergens)
